### PR TITLE
🎉Source Intercom: increase unit test coverage at least 90%

### LINKED ISSUE #11364: Source Intercom: increase unit test coverage at least 90%
## Tell us about the problem you're trying to solve
Need to improve unit test coverage at least 90%

### LINKED ISSUE #11366: Source Intercom: updated doc for OAuth
## Tell us about the problem you're trying to solve
We have the OAuth implemented on the cloud, but there’s no docs update on the website.

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
 #11364 - Source Intercom: increase unit test coverage at least 90%

## How
Increased unit test coverage to 90%

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

</details>

### CODE REVIEW

File: docs/integrations/sources/intercom.md [hunk 1]
```diff
@@ -43,9 +42,13 @@ The Intercom connector should not run into Intercom API limitations under normal
 
 ## Getting started
```
Comment (sherifnada) [inline docs/integrations/sources/intercom.md:43]: this should have a cloud and OSS section, see salesforce docs for an examplee

Comment (lazebnyi) [reply docs/integrations/sources/intercom.md]: Updated.

File: docs/integrations/sources/intercom.md [hunk 2]
```diff
@@ -41,11 +40,13 @@ The connector is restricted by normal Intercom [requests limitation](https://dev
 
 The Intercom connector should not run into Intercom API limitations under normal usage. Please [create an issue](https://github.com/airbytehq/airbyte/issues) if you see any rate limit issues that are not automatically retried successfully.
 
-## Getting started
+## Getting Started (Airbyte Cloud)
 
-### Requirements
+### Setup guide
+
+Please read [How authenticate with OAuth](https://developers.intercom.com/building-apps/docs/authentication-types#section-o-auth)
```
Comment (sherifnada) [inline docs/integrations/sources/intercom.md:47]: that's not quite right, in airbyte cloud they need to click the auth button. We already set it up for them

Comment (lazebnyi) [reply docs/integrations/sources/intercom.md]: Updated to review

**Review by bazarnov — APPROVED** [review]

**Review by davydov-d — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]

**Pull request merged**